File: Computation/Rosalind/DNA.rs
use std::collections::HashMap;
use std::fs;

fn main() {
    let data = fs::read_to_string("./data/rosalind_dna.txt").expect("no file");

    let mut d: HashMap<char, i64> = HashMap::new();

    data.chars().for_each(|c| {
        d.entry(c).and_modify(|v| *v += 1).or_insert(1);
    });

    println!(
        "{}",
        ['A', 'C', 'G', 'T']
            .into_iter()
            .map(|c| d.get(&c).unwrap().to_string())
            .collect::<Vec<String>>()
            .join(" ")
    );
}

File: Computation/Rosalind/DNA.scala
open class Nucleotide:
  var A: Int = 0
  var C: Int = 0
  var G: Int = 0
  var T: Int = 0

  def ToString() =
    s"${A} ${C} ${G} ${T}"

@main def dna_counter() =
  var c = new Nucleotide()

  var data = scala.io.Source
    .fromFile("./data/rosalind_dna.txt")
    .mkString
    .foreach(i =>
      i match
        case 'A' => c.A += 1
        case 'C' => c.C += 1
        case 'G' => c.G += 1
        case 'T' => c.T += 1
        case _   => ()
    )

  println(c.ToString())
